### ISSUE TITLE
JsonQueryFunction always defaults to `rootScope` and fails when alternative Scope is required.

### ISSUE THREAD

**Issue by nickbreentvnz** [eiiches/jackson-jq#17]
When loading all macros, they are all created without specifying a Scope.
https://github.com/eiiches/jackson-jq/blob/0cf9166e8287910334b80e1ced5979ed93477863/jackson-jq/src/main/java/net/thisptr/jackson/jq/Scope.java#L211

Which is a call to: 

https://github.com/eiiches/jackson-jq/blob/0cf9166e8287910334b80e1ced5979ed93477863/jackson-jq/src/main/java/net/thisptr/jackson/jq/internal/JsonQueryFunction.java#L28-L30

... which uses `Scope.rootScope()` which is not guaranteed to be the same scope as the query has been compiled with.

My _hunch_ is that `Scope.loadMacros` just needs to call `new JsonQueryFunction` with `this` as the closure scope, though I've yet to prove it or encode it in a unit test.

```
addFunction(def.name, def.args.size(), new JsonQueryFunction(def.name, def.args, JsonQuery.compile(def.body), this));
```

**Comment by eiiches** [comment]
Thanks. I think you are right.

Fixed in https://github.com/eiiches/jackson-jq/commit/41bda3921849876bd784779939cb1d5d0f9d0678.

**Comment by eiiches** [comment]
Fix released in 0.0.9.
https://github.com/eiiches/jackson-jq/releases/tag/jackson-jq-parent-0.0.9
